File: ParkPal.Application/src/models/api/TimerWithAttraction.ts
import AttractionTimer from "@/models/api/AttractionTimer";
import Attraction from "@/models/api/Attraction";
import Park from "@/models/api/Park";

export default class TimerWithAttraction {
    timer?: AttractionTimer;
    attraction?: Attraction;
    park?: Park;

    constructor(data: Pick<TimerWithAttraction, "timer" | "attraction" | "park"> | null = null) {
        if(data != null) {
            this.timer = new AttractionTimer(data.timer);
            this.attraction = new Attraction(data.attraction);
            this.park = new Park(data.park);
        }
    }
}

File: ParkPal.API/ParkPal.API/Models/Responses/Notification/TimerWithAttraction.cs
using ParkPal.Common.API.Models.ThemeParkApi;
using ParkPal.Common.Models;
using ParkPal.Common.Models.Database.Entities.Notification;

namespace ParkPal.API.Models.Responses.Notification;

public class TimerWithAttraction
{
    public AttractionTimer Timer { get; set; }
    public Attraction Attraction { get; set; }
    public Park Park { get; set; }
}